### PR TITLE
Spyglass: add a link to the PR for presubmits

### PR DESCRIPTION
**Pull request by mirandachrist**
/assign @cjwagner

### CODE REVIEW

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -922,9 +922,11 @@ lensesLoop:
 	}
 
 	prHistLink := ""
+	prLink := ""
 	org, repo, number, err := sg.RunToPR(src)
 	if err == nil {
 		prHistLink = "/pr-history?org=" + org + "&repo=" + repo + "&pr=" + strconv.Itoa(number)
+		prLink = fmt.Sprintf("https://github.com/%s/%s/pull/%d", org, repo, number)
```
Comment (cjwagner) [inline prow/cmd/deck/main.go]: This won't work for GitHub enterprise or gerrit. I think we want to use this link instead: https://github.com/kubernetes/test-infra/blob/6a85e1e9718d1d43aaa4b756826b36e8d22eff71/prow/apis/prowjobs/v1/types.go#L592-L593
Ref: https://github.com/kubernetes/test-infra/pull/10891

File: prow/cmd/deck/template/spyglass.html [hunk 2]
```diff
@@ -23,6 +23,7 @@
     {{if .JobHistLink}}<a href="{{.JobHistLink}}">Job History</a>{{end}}
     {{if .ProwJobLink}}<a href="{{.ProwJobLink}}" onclick="gtag('event', 'view_job_yaml', {event_category: 'engagement', transport_type: 'beacon'})">Prow Job YAML</a>{{end}}
     {{if .PRHistLink}}<a href="{{.PRHistLink}}">PR History</a>{{end}}
+    {{if .PRLink}}<a href="{{.PRLink}}">PR</a>{{end}}
```
Comment (cjwagner) [inline prow/cmd/deck/template/spyglass.html]: You should add `.PRLink` to the conditional on line 21.

File: prow/cmd/deck/main.go [hunk 3]
```diff
@@ -936,6 +936,14 @@ lensesLoop:
 		return "", fmt.Errorf("error determining jobName / buildID: %v", err)
 	}
 
+	prLink := ""
+	j, err := sg.JobAgent.GetProwJob(jobName, buildID)
+	if err != nil {
```
Comment (fejta) [inline prow/cmd/deck/main.go]: is this supposed to be `err == nil`?

File: prow/cmd/deck/main.go [hunk 4]
```diff
@@ -936,6 +936,14 @@ lensesLoop:
...
+		if j.Spec.Refs != nil && len(j.Spec.Refs.Pulls) > 0 {
```
Comment (fejta) [inline prow/cmd/deck/main.go]: combine with previous `if` statement

File: prow/cmd/deck/main.go [hunk 5]
```diff
@@ -936,6 +936,12 @@ lensesLoop:
 		return "", fmt.Errorf("error determining jobName / buildID: %v", err)
 	}
 
+	prLink := ""
+	j, err := sg.JobAgent.GetProwJob(jobName, buildID)
+	if err == nil && j.Spec.Refs != nil && len(j.Spec.Refs.Pulls) > 0 {
```
Comment (fejta) [inline prow/cmd/deck/main.go]: Do we want to log (info??) if there's an error?

**Pull request merged**